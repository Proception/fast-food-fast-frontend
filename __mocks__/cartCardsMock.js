import yam from '../assets/images/yam.jpg';
import rice from '../assets/images/rice.jpg';
import fufu from '../assets/images/fufu.jpg';
import { removeItemFromCart, updateCartTotal, updateItemQuantity, makeOrder } from '../src/actions/orderActions';

const props = {
  item: [
    {
      "id": 1,
      "imageUrl": yam,
      "price": 500,
      "title": "Boiled Yam",
    },
    {
      "id": 2,
      "imageUrl": rice,
      "price": 900,
      "title": "Rice",
    },
    {
      "id": 3,
      "imageUrl": fufu,
      "price": 500,
      "title": "Fufu",
    },
  ],
  onClick: (e) => {},
  onChange: () => {},
  order: {
    cart: [
      {
        "id": 1,
        "imageUrl": yam,
        "price": 500,
        "title": "Boiled Yam",
      },
      {
        "id": 2,
        "imageUrl": rice,
        "price": 900,
        "title": "Rice",
      },
      {
        "id": 3,
        "imageUrl": fufu,
        "price": 500,
        "title": "Fufu",
      },
    ],
    cartTotalPrice: 0
  },
  auth: {
    isLoading: true,
    user: {
      id: 1
    }
  },
  removeItemFromCart,
  updateCartTotal,
  updateItemQuantity,
  makeOrder
};

const emptyCartProps = {
  item: [
    {
      "id": 1,
      "imageUrl": yam,
      "price": 500,
      "title": "Boiled Yam",
    },
    {
      "id": 2,
      "imageUrl": rice,
      "price": 900,
      "title": "Rice",
    },
    {
      "id": 3,
      "imageUrl": fufu,
      "price": 500,
      "title": "Fufu",
    },
  ],
  onClick: (e) => {},
  onChange: () => {},
  order: {
    cart: [],
    cartTotalPrice: 0
  },
  auth: {
    isLoading: false,
    user: {
      id: 1
    }
  },
  removeItemFromCart,
  updateCartTotal,
  updateItemQuantity,
  makeOrder
};
export { props, emptyCartProps };
  