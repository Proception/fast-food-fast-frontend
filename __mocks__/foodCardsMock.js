import yam from '../assets/images/yam.jpg';
import rice from '../assets/images/rice.jpg';
import fufu from '../assets/images/fufu.jpg';

const props = {
  item: [
    {
      "id": 1,
      "imageUrl": yam,
      "price": 500,
      "title": "Boiled Yam",
    },
    {
      "id": 2,
      "imageUrl": rice,
      "price": 900,
      "title": "Rice",
    },
    {
      "id": 3,
      "imageUrl": fufu,
      "price": 500,
      "title": "Fufu",
    },
  ],
  onClick: () => {},
  onChange: () => {}
};
export default props;
  