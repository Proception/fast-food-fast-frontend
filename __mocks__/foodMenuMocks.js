import { addItemToCart, updateCartTotal } from '../src/actions/orderActions';

const props = {
    addItemToCart,
    updateCartTotal,
    auth: {
      user: {
        id:1
      }
    }
    };
const emptyOrderProps = {
    updateCartTotal,
    addItemToCart,
    auth: {
    }
  };
export { props, emptyOrderProps };