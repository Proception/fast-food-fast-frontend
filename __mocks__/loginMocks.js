import { loginUser } from '../src/actions/authActions'

const props = {
    loginUser,
    auth: {
        isLoading: false,
    },
  };
  const propsIsLoading = {
    loginUser,
    auth: {
        isLoading: true,
    },
  };
  const state = {
    loginData: {
        email: 'omasan.esimaje',
        password: 'pass',
    }
  };
export { props, state, propsIsLoading };

