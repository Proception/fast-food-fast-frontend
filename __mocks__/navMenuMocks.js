import { setToken } from '../src/actions/authActions';

const stateGuest = {
    data: [],
    auth: {
        isAuthenticated: false,
    }
  };
  const stateUser = {
    data: [],
    auth: {
        isAuthenticated: true,
        user: {
          email: "omasan.esimaje@gmail.com",
          // full_name: "benedict"
      }
    },
    setToken: setToken,
  };
  const token = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MzEsImZ1bGxfbmFtZSI6IkJlbmVkaWN0Iiwicm9sZV9pZCI6MjAwLCJpYXQiOjE1NTEzODU0MzAsImV4cCI6MTU1MTQ3MTgzMH0.gjqD02e3nwn58TtlNgUmMNMZCVPrdJZ5AGc4hJUslHs'
  export { stateGuest, stateUser, token };