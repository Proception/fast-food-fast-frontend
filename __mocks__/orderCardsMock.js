import { getOrders, getUserOrders } from '../src/actions/orderActions';
const props = {
    orders: [
        {
          "id": 1,
          "orderNo": "12",
          "price": 5000,
          "orderDate": "30 August, 2016",
          "orderStatus": "Pending",
          "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
        },
        {
          "id": 2,
          "orderNo": "54",
          "price": 1200,
          "orderDate": "Boiled Yam",
          "orderStatus": "In Progress",
          "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
        },
        {
          "id": 3,
          "orderNo": "67",
          "price": 1500,
          "orderDate": "30 October, 2017",
          "orderStatus": "Fulfilled",
          "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
        },
        {
          "id": 4,
          "orderNo": "33",
          "price": 3000,
          "orderDate": "30 January, 2018",
          "orderStatus": "Completed",
          "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
        },
      ],
    order: {
      orders: [{
        "id": 1,
        "orderNo": "12",
        "price": 5000,
        "orderDate": "30 August, 2016",
        "orderStatus": "Pending",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
      {
        "id": 2,
        "orderNo": "54",
        "price": 1200,
        "orderDate": "Boiled Yam",
        "orderStatus": "In Progress",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },],
      userOrders: [{
        "id": 1,
        "orderNo": "12",
        "price": 5000,
        "orderDate": "30 August, 2016",
        "orderStatus": "Pending",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
      {
        "id": 2,
        "orderNo": "54",
        "price": 1200,
        "orderDate": "Boiled Yam",
        "orderStatus": "In Progress",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },]
    },
    getOrders,
    getUserOrders
    };
const emptyOrderProps = {
  orders: [
      {
        "id": 1,
        "orderNo": "12",
        "price": 5000,
        "orderDate": "30 August, 2016",
        "orderStatus": "Pending",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
      {
        "id": 2,
        "orderNo": "54",
        "price": 1200,
        "orderDate": "Boiled Yam",
        "orderStatus": "In Progress",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
      {
        "id": 3,
        "orderNo": "67",
        "price": 1500,
        "orderDate": "30 October, 2017",
        "orderStatus": "Fulfilled",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
      {
        "id": 4,
        "orderNo": "33",
        "price": 3000,
        "orderDate": "30 January, 2018",
        "orderStatus": "Completed",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
    ],
  order: {
    orders: [],
    userOrders: [],
  },
  getOrders,
  getUserOrders
  };
export { props, emptyOrderProps };