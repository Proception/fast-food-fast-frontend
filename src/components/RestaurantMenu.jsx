import React, { Component } from 'react';
import { connect } from 'react-redux';
import { addItemToCart, updateCartTotal } from '../actions/orderActions';
import FoodCards from './FoodCards.jsx'
import yam from '../assets/images/yam.jpg';
import rice from '../assets/images/rice.jpg';
import fufu from '../assets/images/fufu.jpg';
import beans from '../assets/images/beans.jpg';

export class RestaurantMenu extends Component {
  state = {
    menu: [ 
      {
        "id": 1,
        "imageUrl": yam,
        "price": 500,
        "title": "Boiled Yam",
        "quantity": 1,
      },
      {
        "id": 2,
        "imageUrl": rice,
        "price": 900,
        "title": "Rice",
        "quantity": 1,
      },
      {
        "id": 3,
        "imageUrl": fufu,
        "price": 500,
        "title": "Fufu",
        "quantity": 1,
      },
      {
        "id": 4,
        "imageUrl": beans,
        "price": 300,
        "title": "Beans",
        "quantity": 1,
      },
    ],
  };

  handleClick = (e, data) => {
    const { addItemToCart, updateCartTotal } = this.props;
    addItemToCart(data);
    updateCartTotal();
  }

  generateMenu(){
    const { menu } = this.state;
    return (
      <ul className="list-group list-group-flush">
      {menu.map(item => (
          <FoodCards key={item.id} onClick={ ((e) => this.handleClick(e, item)) }  item={item}/>
          ))}
      </ul>
      )
  }
  render() {
    return (
      <section className="order">
        <div className="container">
        <div className="header">
            <h3>Core Dishes</h3>
            <div className="pagination">
              <a href="#" >&laquo;</a>
              <a href="#" className="active">1</a>
              <a href="#">2</a>
              <a href="#">&raquo;</a>
            </div>
          </div>

          {
            this.generateMenu()
          }
        </div>
		</section>
    )
  }
}

const mapStateToProps = state => ({
  auth: state.auth,
  order: state.order,
});

export default connect(mapStateToProps, { addItemToCart, updateCartTotal})(RestaurantMenu);
