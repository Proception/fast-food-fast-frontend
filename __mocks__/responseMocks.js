import { token } from './navMenuMocks';

export const okResponse = {
    status: 200,
    response: {
        code: 200,
        status: 'ok',
        messages: 'User logged in successfully',
        data: token
    }
};

export const unauthorizedResponse = {
    status: 401,
    response: {
        code: 401,
        status: 'ok',
        messages: 'User is not authorized',
        data: token
    }
};

export const createdResponse = {
    status: 201,
    response: {
        code: 201,
        status: 'Created',
        messages: 'User signed up in successfully',
        data: token
    }
};

export const noContentResponse = {
    status: 204,
    response: {
        code: 204,
        status: 'Created',
        messages: 'User signed up in successfully',
        data: token
    }
};

export const conflictResponse = {
    status: 409,
    response: {
        code: 409,
        status: 'Conflict',
        messages: 'User exists on the platform',
        data: token
    }
}

export const badRequestResponse = {
    status: 400,
    response: {
        code: 400,
        status: 'Bad Request',
        messages: 'Invalid credentials',
        data: token
    }
}
