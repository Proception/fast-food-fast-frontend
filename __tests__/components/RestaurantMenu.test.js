import React from 'react';
import { shallow } from 'enzyme';
import { RestaurantMenu } from '../../src/components/RestaurantMenu.jsx';
import { props, emptyOrderProps } from '../../__mocks__/foodMenuMocks';
import { preventDefault } from '../../__mocks__/loginMocks';

let wrapper;
beforeEach(() => {
  wrapper = shallow(<RestaurantMenu { ...props }/>);
});
describe('RestaurantMenu Component', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('calls handleClick(e)', () => {
    wrapper = shallow(<RestaurantMenu { ...props }/>);
    const data = {
      "id": 1,
      "imageUrl": 'yam',
      "price": 500,
      "title": "Boiled Yam",
    };
    sinon.spy(wrapper.instance(), 'handleClick');

    wrapper.instance().handleClick(preventDefault, data);

    expect(wrapper.instance().handleClick.calledOnce)
      .toEqual(true);
    expect(wrapper.instance().handleClick.calledWith(preventDefault, data));
  });
  it('calls handleClick(e) guest user', () => {
    wrapper = shallow(<RestaurantMenu { ...emptyOrderProps }/>);
    const data = {
      "id": 1,
      "imageUrl": 'yam',
      "price": 500,
      "title": "Boiled Yam",
    };
    sinon.spy(wrapper.instance(), 'handleClick');

    wrapper.instance().handleClick(preventDefault, data);

    expect(wrapper.instance().handleClick.calledOnce)
      .toEqual(true);
    expect(wrapper.instance().handleClick.calledWith(preventDefault, data));
  });
});
