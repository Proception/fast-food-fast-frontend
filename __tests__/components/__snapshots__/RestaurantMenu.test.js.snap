// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantMenu Component should match snapshot 1`] = `
<section
  className="order"
>
  <div
    className="container"
  >
    <div
      className="header"
    >
      <h3>
        Core Dishes
      </h3>
      <div
        className="pagination"
      >
        <a
          href="#"
        >
          «
        </a>
        <a
          className="active"
          href="#"
        >
          1
        </a>
        <a
          href="#"
        >
          2
        </a>
        <a
          href="#"
        >
          »
        </a>
      </div>
    </div>
    <FoodCards
      menu={
        Array [
          Object {
            "id": 1,
            "imageUrl": "test-file-stub",
            "price": 500,
            "title": "Boiled Yam",
          },
          Object {
            "id": 2,
            "imageUrl": "test-file-stub",
            "price": 900,
            "title": "Rice",
          },
          Object {
            "id": 3,
            "imageUrl": "test-file-stub",
            "price": 500,
            "title": "Fufu",
          },
          Object {
            "id": 4,
            "imageUrl": "test-file-stub",
            "price": 300,
            "title": "Beans",
          },
        ]
      }
    />
  </div>
</section>
`;
